### DISCUSSION TITLE
Use blueretro (ESP32) as transmitter

### DISCUSSION THREAD

**Discussion by almeidarvmware** [darthcloud/BlueRetro#349] [Q&A]
Hello! I couldn't find anything related to this so here I go: 
Due to a lack of good parallel N64 controllers and the abusive prices of the NSO 64 controllers, I found my self thinking about plugging 4 original N64 controllers into a ESP32 and some how "transform" all these controllers into bluetooth devices connecting to blueretro. Is this a dumb idea? Has anyone tried this before?

**Comment by kleffa** [comment]
There is something call ESP32 BLE gamepad https://youtu.be/dODmsoAu0D4
I have seen several mooders that uses it and a lot of solder to use the controller PCB component to make it

**Reply by almeidarvmware** [reply]
> There is something call ESP32 BLE gamepad https://youtu.be/dODmsoAu0D4 I have seen several mooders that uses it and a lot of solder to use the controller PCB component to make it

Thanks for your reply! The guy on this video uses this library https://github.com/lemmingDev/ESP32-BLE-Gamepad and it seems promising. Now I will try to research a way to connect at least one controller to ESP32 and map it to the BLE lib

**Reply by Roet-Ivar** [reply]
> There is something call ESP32 BLE gamepad https://youtu.be/dODmsoAu0D4 I have seen several mooders that uses it and a lot of solder to use the controller PCB component to make it

Hi. I made an ESP32-BLE-Gamepad and I can pair up with blueretro, what button mapping is  blueretro expecting? Right now nothing lines up in terms of mapping. 
@almeidarvmware any pointers regarding mapping?
